client/src/routes/accounts.lazy.tsx

```diff
@@ -54,30 +62,108 @@ function AccountsPage() {
     signupData
   );
 
+  // Generalized change handler for both forms
   const handleChange = (name: string, value: string | boolean) => {
     if (intent === "register") {
+      // Update signup form data
       setSignupData((prevData) => ({
         ...prevData,
         [name]: value,
       }));
     } else {
+      // Update login form data
       setFormData((prevData) => ({
         ...prevData,
         [name]: value,
       }));
     }
   };
 
+  // Function to validate signup form data
+  const validateForm = (formData: typeof signupData) => {
+    const errors: { [key: string]: string } = {};
+    const nameRegex = /^[A-Za-z\s]+$/;
+
+    // Validate first and last name (letters only)
+    if (!nameRegex.test(formData.firstName.trim())) {
+      errors.firstName = "First name must contain only letters";
+    }
+    if (!nameRegex.test(formData.lastName.trim())) {
+      errors.lastName = "Last name must contain only letters";
+    }
+
+    // Validate email format
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(formData.email.trim())) {
+      errors.email = "Invalid email format";
+    }
+
+    // Validate birthdate - check if it's a valid date and if the user is at least 18
+    const birthdate = new Date(formData.birthdate);
+    if (isNaN(birthdate.getTime())) {
+      errors.birthdate = "Invalid birthdate";
+    } else {
+      const today = new Date();
+      const age = today.getFullYear() - birthdate.getFullYear();
+      if (age < 18 || (age === 18 && today < new Date(birthdate.setFullYear(today.getFullYear())))) {
+        errors.birthdate = "Invalid birthdate. You must be at least 18 years old";
+      }
+    }
+
+    // Validate password length
+    if (formData.password.trim().length < 8) {
```

Comment:
There's a small discrepancy here as in the backend the min lenght is set as 6 characters and in the frontend at 8. We might want to adjust that to be the same so we don't get inconsistent data in the future